src/ir_Eurom.cpp

```diff
@@ -0,0 +1,484 @@
+/// @param[in] state The source state to verify the checksum of.
+/// @param[in] length The size of the supplied state.
+/// @return A boolean indicating if its checksum is valid.
+bool IREuromAc::validChecksum(const uint8_t state[], const uint16_t length) {
+  return state[length - 1] == IREuromAc::calcChecksum(state, length);
+}
+
+/// Update the checksum value for the current internal state.
+void IREuromAc::checksum(void) {
+  _.Checksum = IREuromAc::calcChecksum(_.raw, kEuromStateLength);
+}
+
+/// Set the raw state of the remote.
+/// @param[in] state The raw state from the native IR message.
+void IREuromAc::setRaw(const uint8_t state[]) {
+  std::memcpy(_.raw, state, kEuromStateLength);
+}
+
+/// Get the raw state of the remote, suitable to be sent with the appropriate
+/// IRsend object method.
+/// @return A PTR to the internal state.
+uint8_t *IREuromAc::getRaw(void) {
+  checksum();  // Let's ensure this is updated before returning
+  return _.raw;
+}
+
+/// Set the internal state to powered on.
+void IREuromAc::on(void) {
+  setPower(true);
+}
+
+/// Set the internal state to powered off.
+void IREuromAc::off(void) {
+  setPower(false);
+}
+
+/// Set the internal state to use the desired power setting.
+/// @param[in] state The desired power setting.
+void IREuromAc::setPower(const bool state) {
+  // We'll also have to preserve the swing state
+  if (state)
+    _.Power_Swing |= kEuromPowerOn;
+  else
+    _.Power_Swing &= kEuromSwingOn;
+}
+
+/// Get the current power setting from the internal state.
+/// @return A boolean indicating the current power setting.
+bool IREuromAc::getPower(void) const {
+  return (_.Power_Swing & kEuromPowerOn) == kEuromPowerOn;
+}
+
+/// Set the internal state to use the desired operation mode.
+/// @param[in] mode The desired operation mode.
+void IREuromAc::setMode(const uint8_t mode) {
+  switch (mode) {
+    case kEuromCool:
+    case kEuromHeat:
+      state_mode_ = mode;
+      _.Mode_Celsius = getModeCelsiusByte(mode, state_celsius_);
+      break;
+    case kEuromDehumidify:
+    case kEuromVentilate:
+      state_mode_ = mode;
+      _.Mode_Celsius = mode;
+      break;
+    default:
+      break;
+  }
+}
+
+/// Get the current operation mode setting from the internal state.
+/// @return The current operation mode.
+uint8_t IREuromAc::getMode(void) const {
+  return state_mode_;
+}
+
+/// Set the internal state to use the desired temperature.
+/// @param[in] degrees The desired temperature in degrees, normally Celsius.
+/// @param[in] fahrenheit If the given temperature is in Fahrenheit instead.
+void IREuromAc::setTemp(const uint8_t degrees, const bool fahrenheit) {
+  if (state_mode_ != kEuromCool && state_mode_ != kEuromHeat)
+    return;
+
+  uint8_t temp_c, temp_f;
+  if (fahrenheit) {
+    temp_f = std::max(kEuromMinTempF, degrees);
+    temp_f = std::min(kEuromMaxTempF, temp_f);
+    temp_c = static_cast<uint8_t>(std::round(fahrenheitToCelsius(temp_f)));
```

Comment:
This seems to work fine locally, but the pipeline doesn't like `std::round`. What is the preferred method of achieving a rounded value for this library?

Reply:
Not sure why it doesn't like `std::round`.
No real preference, have a look at other similar code.